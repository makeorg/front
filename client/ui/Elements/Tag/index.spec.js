import React from 'react';
import renderer from 'react-test-renderer';
import snapshotDiff from 'snapshot-diff';
import { Tag } from './index';

jest.mock('./Styled', () => ({
  TagStyle: 'TagStyle',
  TagButtonStyle: 'TagButtonStyle',
  TagIconStyle: 'TagIconStyle',
}));

jest.mock('Client/ui/Svg/elements', () => ({
  SvgClose: 'SvgClose',
}));

describe('ComponentStyle', () => {
  it('snapshot by default', () => {
    const component = renderer.create(<Tag name="Hello" />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('must match the diff snapshot between default vs Button Tag', () => {
    const defaultTag = renderer.create(<Tag name="Hello" />).toJSON();
    const buttonTag = renderer.create(<Tag name="Hello" isAButton />).toJSON();
    expect(snapshotDiff(defaultTag, buttonTag)).toMatchSnapshot();
  });

  it('must match the diff snapshot between Button vs Selected Button Tag', () => {
    const buttonTag = renderer.create(<Tag name="Hello" isAButton />).toJSON();
    const selectedButtonTag = renderer
      .create(<Tag name="Hello" isAButton isSelected />)
      .toJSON();
    expect(snapshotDiff(buttonTag, selectedButtonTag)).toMatchSnapshot();
  });
});
