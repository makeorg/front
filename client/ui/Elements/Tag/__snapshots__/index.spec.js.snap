// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentStyle must match the diff snapshot between Button vs Selected Button Tag 1`] = `
Snapshot Diff:
- First value
+ Second value

  <TagStyle
    as="TagButtonStyle"
-   className=""
+   className="selected"
  >
    Hello
  </TagStyle>
`;

exports[`ComponentStyle must match the diff snapshot between default vs Button Tag 1`] = `
Snapshot Diff:
- First value
+ Second value

  <TagStyle
-   as="TagStyle"
+   as="TagButtonStyle"
    className=""
  >
    Hello
  </TagStyle>
`;

exports[`ComponentStyle snapshot by default 1`] = `
<TagStyle
  as="TagStyle"
  className=""
>
  Hello
</TagStyle>
`;
